plugins/module_utils/network/sonic/config/mirroring/mirroring.py

```diff
@@ -0,0 +1,487 @@
+        if commands and len(requests) > 0:
+            if not self._module.check_mode:
+                try:
+                    edit_config(self._module, to_request(self._module, requests))
+                except ConnectionError as exc:
+                    self._module.fail_json(msg=str(exc), code=exc.code)
+            result['changed'] = True
+        result['commands'] = commands
+
+        changed_mirroring_facts = self.get_mirroring_facts()
+
+        result['before'] = existing_mirroring_facts
+        if result['changed']:
+            result['after'] = changed_mirroring_facts
+
+        new_config = changed_mirroring_facts
+        old_config = existing_mirroring_facts
+        if self._module.check_mode:
+            result.pop('after', None)
+            new_config = get_new_config(commands, old_config,
+                                        TEST_KEYS_generate_config)
+            new_config = remove_empties(new_config)
+            result['after(generated)'] = new_config
+
+        if self._module._diff:
+            self.sort_mirrors(old_config)
+            self.sort_mirrors(new_config)
+            result['diff'] = get_formatted_config_diff(old_config,
+                                                       new_config,
+                                                       self._module._verbosity)
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_mirroring_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+
+        have = existing_mirroring_facts
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        state = self._module.params['state']
+        new_want = self.preprocess_want(want)
+        self.validate_want(new_want, state)
+
+        commands = []
+        requests = []
+        if not new_want:
+            new_want = {}
+
+        diff = get_diff(new_want, have, TEST_KEYS)
+
+        if state == 'overridden':
+            commands, requests = self._state_overridden(new_want, have, diff)
+        elif state == 'replaced':
+            commands, requests = self._state_replaced(new_want, have, diff)
+        elif state == 'deleted':
+            commands, requests = self._state_deleted(new_want, have, diff)
```

Comment:
The "diff" argument is not used by the _state_deleted function. It should be removed.